lib/BuildSystem/BuildSystem.cpp

```diff
@@ -730,41 +779,13 @@ class ProducedNodeTask : public Task {
 class DirectoryContentsTask : public Task {
   std::string path;
 
-  /// The exclusion filters used while computing the signature
-  StringList filters;
-
   /// The value for the input directory.
-  ValueType directoryValue;
+  BuildValue directoryValue;
   
   virtual void start(BuildEngine& engine) override {
     // Request the base directory node -- this task doesn't actually use the
     // value, but this connects the task to its producer if present.
 
-    // <rdar://41142590>
-    // FIXME: Temporarily revert part of PR #320, which causes tasks to execute
-    // in the wrong order under certain circumstances.
-    //
-    // The taskMustFollow method expresses the weak dependency we have on
-    // 'path', but only at the task level. What we really want is to say at the
-    // 'isResultValid'/scanning level is 'must scan after'. That way we hold up
-    // this and downstream rules until the 'path' node has been set into its
-    // final state*.
-    //
-    // With the explicit dependency we are establishing with taskNeedsInput, we
-    // will unfortunately mark directory contents as 'needs to be built' under
-    // situations where excluded content and/or non-releveant stat info has
-    // changed. This causes unnecessary rebuilds. See rdar://problem/30640904
-    //
-    // * The 'final state' of a directory is also an thorny patch of toxic land
-    // mines. We really want directory contents to weakly depend upon anything
-    // that is currently and/or may be altered within it. i.e. if one rule
-    // creates the directory and another rule writes a file into it, we want to
-    // defer scanning until both of them have been scanned and possibly run.
-    // Having a 'must scan after' would help with the first rule (mkdir), but
-    // not the second, in particular if rules are added in subsequent builds.
-    // Related rdar://problem/30638921
```

Comment:
Are you sure it is worth deleting this entire comment? Some parts of it remain true.